compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp

```diff
@@ -49,50 +53,67 @@ static SmallVector<NamedSequenceOp> findTuningSpecs(ModuleOp module) {
+// ```
+// transform.named_sequence @name(%arg0: !transform.any_op) ->
+// (!transform.any_op)
+// ```
 static LogicalResult validateTuningSpec(NamedSequenceOp op) {
-  if (!op.getResultTypes().empty()) {
-    op->emitWarning() << "Tuning spec expected to have no results";
-    return failure();
+  ArrayRef<Type> resTypes = op.getFunctionType().getResults();
+  if (resTypes.size() != 1 || !isa<transform::AnyOpType>(resTypes[0])) {
+    return op.emitWarning()
+           << "Tuning spec entry point expected to return any_op";
   }
 
   ArrayRef<Type> argTypes = op.getArgumentTypes();
   if (argTypes.size() != 1 || !isa<transform::AnyOpType>(argTypes[0])) {
-    op->emitWarning() << "Tuning spec expected to have one argument of type "
-                         "'!transform.any_op'";
-    return failure();
-  }
-
-  if (!op.getArgAttr(0, transform::TransformDialect::kArgReadOnlyAttrName)) {
-    op->emitWarning() << "Tuning spec expected to have one readonly argument";
-    return failure();
+    return op.emitWarning() << "Tuning spec entry point expected to have a "
+                               "single any_op argument";
   }
 
   return success();
 }
 
+static bool consumesInputOp(NamedSequenceOp op) {
+  if (op.getArgAttr(0, kArgConsumedAttrName)) {
+    return true;
+  }
+  return false;
+}
+
 static NamedSequenceOp
 emitLinkedTuningSpec(ModuleOp module, ArrayRef<NamedSequenceOp> specsToLink) {
   OpBuilder builder(module->getContext());
   builder.setInsertionPointToEnd(module.getBody());
 
+  const bool hasConsumedSequences = llvm::any_of(specsToLink, consumesInputOp);
   Location loc = builder.getFusedLoc(llvm::map_to_vector(
       specsToLink, [](NamedSequenceOp op) { return op->getLoc(); }));
-  FunctionType specType = builder.getFunctionType(
-      TypeRange{builder.getType<transform::AnyOpType>()}, TypeRange{});
+  Type anyOpType = builder.getType<transform::AnyOpType>();
+  FunctionType specType =
+      builder.getFunctionType(TypeRange{anyOpType}, TypeRange{anyOpType});
   auto newSpec = builder.create<NamedSequenceOp>(
       loc, kKernelConfigSpecName, TypeAttr::get(specType),
       /*sym_visibility=*/StringAttr{},
       /*arg_attrs=*/ArrayAttr{},
       /*res_attrs*/ ArrayAttr{});
-  newSpec.setArgAttr(0, transform::TransformDialect::kArgReadOnlyAttrName,
-                     builder.getUnitAttr());
+  newSpec.setArgAttr(
+      0, hasConsumedSequences ? kArgConsumedAttrName : kArgReadOnlyAttrName,
+      builder.getUnitAttr());
   newSpec->setAttr(kTuningSpecEntrypointAttrName, builder.getUnitAttr());
 
   Region &region = newSpec.getRegion();
   Block *body = builder.createBlock(&region, region.begin(),
                                     newSpec.getArgumentTypes(), loc);
   builder.setInsertionPointToStart(body);
 
+  // Make sure spec names are unique to work around a transform dialect
+  // interpreter bug (`transform.include` does not handle name collisions
+  // correctly).
```

Comment:
Can you file an issue and link to it so we can track it?

Reply:
I filed an issue here: https://github.com/llvm/llvm-project/issues/119578